### ISSUE TITLE
ANR on Android 4.4.4 due to high system load

### ISSUE THREAD

**Issue by Catfriend1** [Catfriend1/syncthing-android#637]
### Version Information
    App Version: 1.4.2.2
    Syncthing Version: v1.4.2
    Android Version: Android 4.4.4 / Cyanogenmod
    Device: Samsung N7000 (Note)

### Android Log
```
04-13 10:49:46.570 I/Util    (7346): runShellCommandGetOutput: Exited with code 0
04-13 10:49:46.640 D/Util    (7346): runShellCommandGetOutput: ps
04-13 10:49:46.755 I/Util    (7346): runShellCommandGetOutput: Exited with code 0
04-13 10:49:46.830 D/Util    (7346): runShellCommandGetOutput: ps
04-13 10:49:47.010 I/Util    (7346): runShellCommandGetOutput: Exited with code 0
04-13 10:49:47.095 D/Util    (7346): runShellCommandGetOutput: ps
04-13 10:49:47.220 I/Util    (7346): runShellCommandGetOutput: Exited with code 0
04-13 10:49:47.295 D/Util    (7346): runShellCommandGetOutput: ps
04-13 10:49:47.390 I/Util    (7346): runShellCommandGetOutput: Exited with code 0
04-13 10:49:47.470 D/Util    (7346): runShellCommandGetOutput: ps
04-13 10:49:47.585 I/Util    (7346): runShellCommandGetOutput: Exited with code 0
04-13 10:49:47.665 D/Util    (7346): runShellCommandGetOutput: ps
04-13 10:49:47.870 I/Util    (7346): runShellCommandGetOutput: Exited with code 0
04-13 10:49:47.935 D/Util    (7346): runShellCommandGetOutput: ps
04-13 10:49:48.050 I/Util    (7346): runShellCommandGetOutput: Exited with code 0
04-13 10:49:48.120 D/Util    (7346): runShellCommandGetOutput: ps
04-13 10:49:48.195 I/Util    (7346): runShellCommandGetOutput: Exited with code 0
04-13 10:49:48.275 D/Util    (7346): runShellCommandGetOutput: ps
04-13 10:49:48.395 I/Util    (7346): runShellCommandGetOutput: Exited with code 0
04-13 10:49:48.465 D/Util    (7346): runShellCommandGetOutput: ps
04-13 10:49:48.575 I/Util    (7346): runShellCommandGetOutput: Exited with code 0
04-13 10:49:48.640 D/Util    (7346): runShellCommandGetOutput: ps
04-13 10:49:48.810 I/Util    (7346): runShellCommandGetOutput: Exited with code 0
04-13 10:49:48.875 D/Util    (7346): runShellCommandGetOutput: ps
04-13 10:49:48.995 I/Util    (7346): runShellCommandGetOutput: Exited with code 0
04-13 10:49:49.065 D/Util    (7346): runShellCommandGetOutput: ps
04-13 10:49:49.195 I/Util    (7346): runShellCommandGetOutput: Exited with code 0
04-13 10:49:49.270 D/Util    (7346): runShellCommandGetOutput: ps
04-13 10:49:49.370 I/Util    (7346): runShellCommandGetOutput: Exited with code 0
04-13 10:49:49.440 D/Util    (7346): runShellCommandGetOutput: ps
04-13 10:49:49.620 I/Util    (7346): runShellCommandGetOutput: Exited with code 0
04-13 10:49:49.705 D/Util    (7346): runShellCommandGetOutput: ps
04-13 10:49:49.895 I/Util    (7346): runShellCommandGetOutput: Exited with code 0
04-13 10:49:49.975 D/Util    (7346): runShellCommandGetOutput: ps
04-13 10:49:50.100 I/Util    (7346): runShellCommandGetOutput: Exited with code 0
04-13 10:49:50.140 D/lights  (2274): set_light_buttons: 2
04-13 10:49:50.140 I/InputDispatcher(2274): Application is not responding: Window{42702808 u0 com.github.catfriend1.syncthingandroid.debug/com.nutomic.syncthingandroid.activities.MainActivity}.  It has been 5004.1ms since event, 5003.7ms since wait started.  Reason: Waiting because the touched window has not finished processing the input events that were previously delivered to it.
04-13 10:49:50.155 I/WindowManager(2274): Input event dispatching timed out sending to com.github.catfriend1.syncthingandroid.debug/com.nutomic.syncthingandroid.activities.MainActivity.  Reason: Waiting because the touched window has not finished processing the input events that were previously delivered to it.
04-13 10:49:50.185 D/Util    (7346): runShellCommandGetOutput: ps
04-13 10:49:50.195 D/lights  (2274): set_light_buttons: 1
04-13 10:49:50.225 I/Process (2274): Sending signal. PID: 7346 SIG: 3
04-13 10:49:50.225 I/dalvikvm(7346): threadid=3: reacting to signal 3
04-13 10:49:50.335 I/dalvikvm(7346): Wrote stack traces to '/data/anr/traces.txt'
04-13 10:49:50.335 I/Process (2274): Sending signal. PID: 2274 SIG: 3
04-13 10:49:50.335 I/dalvikvm(2274): threadid=3: reacting to signal 3
04-13 10:49:50.400 I/SyncthingNativeCode(7346): [6WY6O] INFO: Detected 0 NAT services
04-13 10:49:50.405 I/SyncthingNativeCode(7346): [6WY6O] INFO: Exiting
04-13 10:49:50.405 I/Util    (7346): runShellCommandGetOutput: Exited with code 0
04-13 10:49:50.420 I/SyncthingRunnable(7346): Syncthing was shut down normally via API or SIGKILL. Exit code = 0
04-13 10:49:50.475 D/Util    (7346): runShellCommandGetOutput: ps
04-13 10:49:50.575 E/Sensors (2274): poll() failed (Interrupted system call)
04-13 10:49:50.595 I/dalvikvm(2274): Wrote stack traces to '/data/anr/traces.txt'
04-13 10:49:50.600 I/Process (2274): Sending signal. PID: 2585 SIG: 3
04-13 10:49:50.600 I/dalvikvm(2585): threadid=3: reacting to signal 3
04-13 10:49:50.600 I/Process (2274): Sending signal. PID: 2576 SIG: 3
04-13 10:49:50.600 I/dalvikvm(2576): threadid=3: reacting to signal 3
04-13 10:49:50.610 I/Util    (7346): runShellCommandGetOutput: Exited with code 0
04-13 10:49:50.610 D/SyncthingRunnable(7346): killSyncthing: Complete.
04-13 10:49:50.610 D/SyncthingService(7346): Finished mSyncthingRunnableThread.
04-13 10:49:50.615 I/dalvikvm(2585): Wrote stack traces to '/data/anr/traces.txt'
04-13 10:49:50.615 I/Process (2274): Sending signal. PID: 2571 SIG: 3
04-13 10:49:50.615 I/dalvikvm(2571): threadid=3: reacting to signal 3
04-13 10:49:50.625 I/dalvikvm(2576): Wrote stack traces to '/data/anr/traces.txt'
04-13 10:49:50.625 I/Process (2274): Sending signal. PID: 2387 SIG: 3
04-13 10:49:50.630 I/dalvikvm(2387): threadid=3: reacting to signal 3
04-13 10:49:50.665 D/Util    (7346): runShellCommandGetOutput: netstat -t -l -n
04-13 10:49:50.690 I/dalvikvm(2387): Wrote stack traces to '/data/anr/traces.txt'
04-13 10:49:50.710 I/Util    (7346): runShellCommandGetOutput: Exited with code 0
04-13 10:49:50.715 I/dalvikvm(2571): Wrote stack traces to '/data/anr/traces.txt'
04-13 10:49:50.730 I/SyncthingService(7346): onServiceStateChange: from INIT to STARTING
04-13 10:49:50.730 I/SyncthingService(7346): Web GUI will be available at http://127.0.0.1:8384
04-13 10:49:50.755 D/Util    (7346): runShellCommandGetOutput: ps
04-13 10:49:50.845 I/Util    (7346): runShellCommandGetOutput: Exited with code 0
04-13 10:49:50.850 I/PollWebGuiAvailableTask(7346): Starting to poll for web gui availability
04-13 10:49:50.855 I/InputDispatcher(2274): Window 'Window{42702808 u0 com.github.catfriend1.syncthingandroid.debug/com.nutomic.syncthingandroid.activities.MainActivity}' spent 6446.0ms processing the last input event: MotionEvent(deviceId=7, source=0x00001002, action=0, flags=0x00000000, metaState=0x00000000, buttonState=0x00000000, edgeFlags=0x00000000, xPrecision=1.0, yPrecision=1.0, displayId=0, pointers=[0: (444.0, 222.0)]), policyFlags=0x47000000
04-13 10:49:50.855 I/InputDispatcher(2274): Window 'Window{42702808 u0 com.github.catfriend1.syncthingandroid.debug/com.nutomic.syncthingandroid.activities.MainActivity}' spent 6382.2ms processing the last input event: MotionEvent(deviceId=7, source=0x00001002, action=1, flags=0x00000000, metaState=0x00000000, buttonState=0x00000000, edgeFlags=0x00000000, xPrecision=1.0, yPrecision=1.0, displayId=0, pointers=[0: (444.0, 222.0)]), policyFlags=0x47000000
04-13 10:49:50.945 D/Util    (7346): runShellCommand: chmod 500 /data/app-lib/com.github.catfriend1.syncthingandroid.debug-2/libsyncthingnative.so
04-13 10:49:50.950 W/SyncthingRunnable(7346): chmod SyncthingNative failed with exit code 10
04-13 10:49:51.000 V/NotificationHandler(7346): Starting foreground service or updating notification
04-13 10:49:51.005 W/ApiRequest(7346): Request to http://127.0.0.1:8384/rest/svc/report failed, java.net.ConnectException: failed to connect to /127.0.0.1 (port 8384) after 5000ms: isConnected failed: ECONNREFUSED (Connection refused)
04-13 10:49:51.075 V/NotificationHandler(7346): Starting foreground service or updating notification
04-13 10:49:51.085 I/SyncthingNativeCode(7346): [start] INFO: syncthing v1.4.2 "Fermium Flea" (go1.13.9 android-arm) CF\Anwender@NETSRV 2020-04-07 11:14:03 UTC
04-13 10:49:51.185 V/PollWebGuiAvailableTask(7346): Polling web gui ... (1)
04-13 10:49:51.615 I/SyncthingNativeCode(7346): [start] INFO: No automatic upgrades; STNOUPGRADE environment variable defined.
04-13 10:49:51.615 I/SyncthingNativeCode(7346): [6WY6O] INFO: My ID: 6WY6OPP-C6RZKOF-IPF6ZRA-5JKJ66A-SLYUQPS-GJREBJQ-NBHUQJ6-IEHTPQ6
04-13 10:49:51.850 I/Choreographer(7346): Skipped 31 frames!  The application may be doing too much work on its main thread.
04-13 10:49:52.810 I/SyncthingNativeCode(7346): [6WY6O] INFO: Single thread SHA256 performance is 15 MB/s using crypto/sha256 (13 MB/s using minio/sha256-simd).
04-13 10:49:53.420 V/PollWebGuiAvailableTask(7346): Polling web gui ... (10)
04-13 10:49:53.960 I/Process (2274): Sending signal. PID: 6525 SIG: 3
04-13 10:49:53.960 I/dalvikvm(6525): threadid=3: reacting to signal 3
04-13 10:49:54.110 D/dalvikvm(6525): ExtrMon: insns 0x0 not valid (0x0 - 0x0)
04-13 10:49:54.110 D/dalvikvm(6525): ExtrMon: insns 0x0 not valid (0x0 - 0x0)
04-13 10:49:54.170 I/Process (2274): Sending signal. PID: 6962 SIG: 3
04-13 10:49:54.170 I/dalvikvm(6962): threadid=3: reacting to signal 3
04-13 10:49:54.175 I/dalvikvm(6525): Wrote stack traces to '/data/anr/traces.txt'
04-13 10:49:54.180 I/Process (2274): Sending signal. PID: 7346 SIG: 6
04-13 10:49:54.180 E/ActivityManager(2274): ANR in com.github.catfriend1.syncthingandroid.debug (com.github.catfriend1.syncthingandroid.debug/com.nutomic.syncthingandroid.activities.MainActivity)
04-13 10:49:54.180 E/ActivityManager(2274): PID: 7346
04-13 10:49:54.180 E/ActivityManager(2274): Reason: Input dispatching timed out (Waiting because the touched window has not finished processing the input events that were previously delivered to it.)
04-13 10:49:54.180 E/ActivityManager(2274): Load: 4.43 / 4.17 / 3.5
04-13 10:49:54.180 E/ActivityManager(2274): CPU usage from 7708ms to 0ms ago:
04-13 10:49:54.180 E/ActivityManager(2274):   44% 7346/com.github.catfriend1.syncthingandroid.debug: 35% user + 9% kernel / faults: 74539 minor
04-13 10:49:54.180 E/ActivityManager(2274):   31% 7621/libsyncthingnative.so: 18% user + 13% kernel / faults: 804 minor
04-13 10:49:54.180 E/ActivityManager(2274):   9.3% 2387/com.android.systemui: 7% user + 2.3% kernel / faults: 204 minor

```

**Comment by Catfriend1** [comment]
Can't be fixed, just a notice that Android 4.4.4 does kill apps taking too long to load with an ANR in log. Recent Android OS versions let the user decide to wait or terminate the app. So nothing to fix here.

**Comment by Catfriend1** [comment]
@funnyflowerpot I don't think this issue is related to the kernel bug causing hot reboot - it's just ANR not solvable because it happens within the nstive. We've had other bad behaviour of Android 4.4.4 regarding interrupted system calls so as much people might hate me I'm considering the sooner or later discontinuation of 4.x. ( see https://forum.syncthing.net/t/repeated-interrupted-system-call-on-multiple-android-devices/14626/33 for what mean. )
